src/ehrdata/_feature_types.py

```diff
@@ -0,0 +1,326 @@
+                branch.add(f"{unencoded} ({edata.obs[unencoded].nunique()} categories); {enc_mode} encoded")
+
+    else:
+        for categorical in sorted(cat_features):
+            categorical_feature = df.loc[:, categorical]
+            categorical_feature_nans_cleaned = categorical_feature.replace(MISSING_VALUES, np.nan)
+            branch.add(f"{categorical} ({categorical_feature_nans_cleaned.nunique()} categories)")
+
+    print(tree)
+
+
+def replace_feature_types(
+    edata: EHRData,
+    features: Iterable[str],
+    corrected_type: str,
+) -> None:
+    """Correct the feature types for a list of features inplace.
+
+    Args:
+        edata: Data object.
+        features: The features to correct.
+        corrected_type: The corrected feature type. One of `'date'`, `'categorical'`, or `'numeric'`.
+
+    Examples:
+        >>> import ehrdata as ed
+        >>> edata = ed.dt.diabetes_130_fairlearn()
+        >>> ed.infer_feature_types(edata)
+        >>> ed.replace_feature_types(edata, ["time_in_hospital", "number_diagnoses", "num_procedures"], "numeric")
+    """
+    if FEATURE_TYPE_KEY not in edata.var:
+        err_msg = "Feature types were not inferred. Please infer feature types using 'ed.infer_feature_types' before correcting."
+        raise ValueError(err_msg)
+
+    if corrected_type not in [CATEGORICAL_TAG, NUMERIC_TAG, DATE_TAG]:
+        err_msg = f"Corrected type {corrected_type} not recognized. Choose between '{DATE_TAG}', '{CATEGORICAL_TAG}', or '{NUMERIC_TAG}'."
+        raise ValueError(err_msg)
+
+    if FEATURE_TYPE_KEY not in edata.var:
+        err_msg = (
+            "Feature types were not inferred. Please infer feature types using 'infer_feature_types' before correcting."
+        )
+        raise ValueError(err_msg)
+
+    if isinstance(features, str):
+        features = [features]
+
+    edata.var.loc[features, FEATURE_TYPE_KEY] = corrected_type
+
+
+def harmonize_missing_values(
+    edata: EHRData,
+    *,
+    layer: str | None = None,
+    missing_values: Iterable[str] | None = ["nan", "np.nan", "<NA>", "pd.NA"],
+    copy: bool = False,
+) -> EHRData | None:
+    """Harmonize missing values in the :class:`~ehrdata.EHRData` object.
+
+    This function will replace strings that are considered to represent missing values with `np.nan`.
+
+
+    Args:
+        edata: Data object.
+        layer: The layer to use from the :class:`~ehrdata.EHRData` object. If `None`, the `X` layer is used.
+        missing_values: The strings that are considered to represent missing values and should be replaced with np.nan
+        copy: Whether to return a copy of the :class:`~ehrdata.EHRData` object with the missing values replaced.
+
+    Examples:
+        >>> import ehrdata as ed
+        >>> edata = ed.dt.mimic_2()
+        >>> ed.harmonize_missing_values(edata)
+    """
+    if copy:
+        edata = edata.copy()
+    X = edata.X if layer is None else edata.layers[layer]
```

Comment:
We'll need an internal function for this as soon as Phil's PR is merged. Let's not forget about this. You can resolve this comment.